src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirstNode(value);
+        } else {
+            addLastNode(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            addFirstNode(value);
```

Comment:
If you check for isEmpty first and then call checkIndex, you don't have to write size + 1